### DISCUSSION TITLE
Items from JOB_DETAILS are not being removed when job or trigger is deleted

### DISCUSSION THREAD

**Discussion by CrazyMoka** [quartznet/quartznet#2240] [Q&A] [answered]
Hello.
When I run scheduler.Interrupt(_t.JobKey);
scheduler.UnscheduleJob(_tKey);
scheduler.DeleteJob(_t.JobKey);
either of those statements Job_Details table still has data in it.

There was a job in the TRIGGERS table with the same Job name and group. When I hard coded to delete that with DeleteJob they all went away however in JOB_Details information for that job still remained.  

What am I doing wrong?

Regards

**Comment by CrazyMoka** [comment]
Example
Scheduler-Core	TestJob	Basic	NULL	Utilities.TestJob, Utilities	1	1	1	0	NULL


This job stayed when I ran scheduler.DeleteJob(JobKey('TestJob','Basic'));

**Comment by CrazyMoka** [comment]
Also, Cron_TRiggers are not being removed.
Is the schedule.UnscheduleJob supposed to remove those cron triggers?

The jobs in the job_details table keep piling up.. They sometimes delete but not all the time.

**Comment by CrazyMoka** [comment]
I'm starting to think Persisted Jobs need to be removed manually from JOB_DETAILS.
I've removed all triggers using the proper api looping through and trying to delete jobs and they still remain. No other data is in my DB

Not sure why this is happening.

**Reply by lahma** [reply]
If job is stored as durable it will stay in the table even if no triggers refers to it, if not durable jobs should be automatically deleted after last trigger referencing it has been deleted.

**Reply by CrazyMoka** [reply]
So if a job is durable I do indeed need to remove it manually sql statement?
I thought the api would have a way to delete durable jobs.

**Reply by lahma** [reply]
You should be able to delete it with the API, it just won't be automatically deleted when orphaned.

**Reply by CrazyMoka** [reply]
Seems to be a timing thing in my delete method. Thanks

**Reply by lahma** [reply]
You are not `await`ing the API calls?

**Reply by CrazyMoka** [reply, marked as answer]
The issue was I was running Interrupt instead of pause then delete.
